covalent/_workflow/transportable_object.py

```diff
@@ -0,0 +1,420 @@
+
+    @staticmethod
+    def deserialize(
+        serialized: bytes, *, header_only: bool = False, string_only: bool = False
+    ) -> "TransportableObject":
+        """Deserialize the transportable object.
+
+        Args:
+            data: Serialized transportable object
+
+        Returns:
+            object: The deserialized transportable object.
+
+        """
+        ar = _TOArchive.load(serialized, header_only, string_only)
+        return _from_archive(ar)
+
+    @staticmethod
+    def deserialize_list(collection: list) -> list:
+        """
+        Recursively deserializes a list of TransportableObjects. More
+        precisely, `collection` is a list, each of whose entries is
+        assumed to be either a `TransportableObject`, a list, or dict`
+
+        Args:
+            collection: A list of TransportableObjects.
+
+        Returns:
+            A list of deserialized objects.
+
+        """
+        new_list = []
+        for item in collection:
+            if isinstance(item, TransportableObject):
+                new_list.append(item.get_deserialized())
+            elif isinstance(item, list):
+                new_list.append(TransportableObject.deserialize_list(item))
+            elif isinstance(item, dict):
+                new_list.append(TransportableObject.deserialize_dict(item))
+            else:
+                raise TypeError("Couldn't deserialize collection")
+        return new_list
+
+    @staticmethod
+    def deserialize_dict(collection: dict) -> dict:
+        """
+        Recursively deserializes a dict of TransportableObjects. More
+        precisely, `collection` is a dict, each of whose entries is
+        assumed to be either a `TransportableObject`, a list, or dict`
+
+        Args:
+            collection: A dictionary of TransportableObjects.
+
+        Returns:
+            A dictionary of deserialized objects.
+
+        """
+        new_dict = {}
+        for k, item in collection.items():
+            if isinstance(item, TransportableObject):
+                new_dict[k] = item.get_deserialized()
+            elif isinstance(item, list):
+                new_dict[k] = TransportableObject.deserialize_list(item)
+            elif isinstance(item, dict):
+                new_dict[k] = TransportableObject.deserialize_dict(item)
+            else:
+                raise TypeError("Couldn't deserialize collection")
+        return new_dict
+
+
+def _to_archive(to: TransportableObject) -> _TOArchive:
+    """Convert a TransportableObject to a _TOArchive.
+
+    Args:
+        to: Transportable object to be converted.
+
+    Returns:
+        Archived transportable object.
+
+    """
+    header = json.dumps(to._header).encode("utf-8")
+    object_string = to._object_string.encode("utf-8")
+    data = to._object.encode("utf-8")
+    return _TOArchive(header=header, object_string=object_string, data=data)
+
+
+def _from_archive(ar: _TOArchive) -> TransportableObject:
```

Comment:
This would make sense (for consistency) to be `to_transportable_object` static method on the `_TOArchive` class

Reply:
Changed!